pkg/util/resizefs/resizefs_linux.go

```diff
@@ -50,6 +53,19 @@ func (resizefs *ResizeFs) Resize(devicePath string, deviceMountPath string) (boo
 		return false, nil
 	}
 
+	// don't fail if resolving doesn't work
+	if blockDeviceRescanPath, err := findBlockDeviceRescanPath(devicePath); err != nil {
+		klog.V(0).Infof("ResizeFS.Resize - error resolving block device path from %q: %v", devicePath, err)
+	} else {
+		klog.V(3).Infof("ResizeFS.Resize - resolved block device path from %q to %q", devicePath, blockDeviceRescanPath)
+
+		klog.V(3).Infof("ResizeFS.Resize - polling %q block device geometry", devicePath)
+		err = ioutil.WriteFile(blockDeviceRescanPath, []byte{'1'}, 0666)
+		if err != nil {
+			klog.V(0).Infof("ResizeFS.Resize - error polling new block device geometry: %v", err)
+		}
```

Comment:
I do not have a system where I could test this right now but is new size immediately after writing to `/sys/block/xxx/device/rescan` file? As the comment implies does this needs to poll for device geometry before calling ext/xfs resize utils? cc @bswartz

Reply:
I recommend checking the block device size after rescaning and fail if the size hasn't been updated. If the update is synchronous, this is harmless. If the update is asynchronous and slow, then this will succeed during the next retry.

Reply:
I looked at the kernel code and it appears that the rescan is in fact synchronous, but I still think that checking the size is good practice for defense against bugs elsewhere.

Reply:
I don't think that the check has to be here. Let's assume that somewhen the device polling won't be necessary, in this case the device size won't be increased, and the function will fail the old size vs new size comparison check.